File: code-ninja-course/JAVA/src/main/java/com/codeninja/solidsolution/students/Student.java
package com.codeninja.solidsolution.students;

import com.codeninja.solidsolution.PackageType;
import com.codeninja.solidsolution.exceptions.*;

public class Student {
	int universityID;
	String emailAddress;

	public Student(String emailAddress, int universityID) {
			this.universityID = universityID;
			this.emailAddress = emailAddress;
	}

//	public void setAllowance(Student student, PackageType type) {
//		int factor = rules.get(type).getFactor();
//		student.setMonthlyEbookAllowance(factor * 10);
//	}
//
	public void setMonthlyEbookAllowance(int allowence) {
	}

	public String getEmailAddress() {
		throw new NotImplemented();
	}
	public int getUniversityID() {
		throw new NotImplemented();
	}
	int getMonthlyEbookAllowance() {
		throw new NotImplemented();
	}
	int getCurrentlyBorrowedEbooks() {
		throw new NotImplemented();
	}
	void setCurrentlyBorrowedEbooks(int books) {
		throw new NotImplemented();
	}

}

File: code-ninja-course/CSHARP/Code/SOLIDSolution/Students/Student.cs
namespace LegacyCodeToTestableCode.SOLIDSolution
{
    public class Student
    {
        int universityID;
        string emailAddress;

        public Student(string emailAddress, int universityID)
        {
            this.universityID = universityID;
            this.emailAddress = emailAddress;
        }

        public void setMonthlyEbookAllowance(int allowence)
        {
        }

        public string getEmailAddress()
        {
            throw new NotImplemented();
        }

        public int getUniversityID()
        {
            throw new NotImplemented();
        }

        int getMonthlyEbookAllowance()
        {
            throw new NotImplemented();
        }

        int getCurrentlyBorrowedEbooks()
        {
            throw new NotImplemented();
        }

        void setCurrentlyBorrowedEbooks(int books)
        {
            throw new NotImplemented();
        }

    }

}
